### PR TITLE
[15.0][MIG] website_sale_coupon_restrict: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc@Tecnativa TT37333

@chienandalu @CarlosRoca13 please review!

### CODE REVIEW

File: website_sale_coupon_restrict/models/sale_coupon.py [hunk 1]
```diff
@@ -4,9 +4,10 @@
 
 
 class SaleCoupon(models.Model):
-    _inherit = "sale.coupon"
+    _inherit = "coupon.coupon"
 
-    def _check_coupon_code(self, order):
+    def _check_coupon_code(self, order_date, partner_id, **kwargs):
+        order = kwargs.get("order", False)
         if self.program_id.website_only and not order.website_id:
```
Comment (chienandalu) [inline website_sale_coupon_restrict/models/sale_coupon.py]: You should check if order exists as well

**Review by chienandalu — CHANGES_REQUESTED** [review]
<del>As usual, split core and sale functionality</del>

**Comment by pedrobaeza** [comment]
Not sure if in this case it's worth the split, being this totally oriented to e-commerce

**Comment by chienandalu** [comment]
> Not sure if in this case it's worth the split, being this totally oriented to e-commerce

Yes, that's right, we're depending on website_sale_coupon, so the comment isn't relevant.

File: website_sale_coupon_restrict/models/sale_coupon.py (hunk 1 above)
Comment (stefan-tecnativa) [reply website_sale_coupon_restrict/models/sale_coupon.py]: Done, thanks for the suggest

**Review by CarlosRoca13 — APPROVED** [review]
Code review :+1:

File: website_sale_coupon_restrict/views/sale_coupon_program_views.xml [hunk 2]
```diff
@@ -1,12 +1,9 @@
...
 
-    <record model="ir.ui.view" id="sale_coupon_program_view_form_common">
-        <field name="model">sale.coupon.program</field>
-        <field
-            name="inherit_id"
-            ref="sale_coupon.sale_coupon_program_view_form_common"
-        />
+    <record model="ir.ui.view" id="coupon_program_view_form_common">
+        <field name="model">coupon.program</field>
+        <field name="inherit_id" ref="coupon.coupon_program_view_coupon_program_form" />
```
Comment (chienandalu) [inline website_sale_coupon_restrict/views/sale_coupon_program_views.xml]: This view is just for coupon programs, you should inherit the common one.

**Review by chienandalu — CHANGES_REQUESTED** [review]

**Review by chienandalu — APPROVED** [review]

**Comment by chienandalu** [comment]
Ready to merge @pedrobaeza

**Pull request merged**